crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net.rs

```diff
@@ -636,6 +729,20 @@ impl MetaNet {
         evt_sender: futures::channel::mpsc::Sender<KitsuneP2pEvent>,
         signal_url: String,
     ) -> KitsuneP2pResult<(Self, MetaNetEvtRecv)> {
+        opentelemetry_api::global::meter("kitsune_p2p");
```

Comment:
If we're going to put metrics in the code rather than wrapping them like on #2513 then I think establishing naming conventions early on will be necessary. Otherwise these quickly can become hard to write queries for without regular expressions etc.

Similarly doing things like labeling all metrics with an `instance_id` is more difficult if they are spread out. Maybe opentelemetry supports setting global labels. I'm not sure if it does or not.

Reply:
> establishing naming conventions early on will be necessary

I'm all for establishing a convention. Any thoughts?

> Maybe opentelemetry supports setting global labels

[versioned_meter](https://docs.rs/ts_opentelemetry_api/0.20.0-beta.1/ts_opentelemetry_api/metrics/trait.MeterProvider.html#tymethod.versioned_meter) allows attributes. I can experiment with that.

Reply:
Versioned meter looks like the right kind of things, something that doesn't involve passing the attributes on `observe`.

For naming conventions there are several reference ones for databases and some other common technology https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/ along with general guidance.

It would be nice to create a hackmd or similar like this https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/database-metrics/#metric-instruments
So we'd end up with something like `conductor.websocket_connections.max`, `kitsune_p2p.connections.timeouts` and so on.

I don't think we need to write it all down up front, just pick a naming pattern and then when other people come to add new metrics they can at least re-use the name for the area of the system and refer to a best practice guide from OpenTelemetry. Then hopefully update the document as well as the code

Reply:
I *was* able to make the versioned_meter attributes work, though I don't think they thought through the API very well, it's awkward to pass in `None`s to the version / schema url, since it doesn't know the types, and you have to go through the `meter_provider()` global accessor:

```rust
    let meter = opentelemetry_api::global::meter_provider().versioned_meter(
        "my_metrics",
        None::<&'static str>,
        None::<&'static str>,
        Some(vec![opentelemetry_api::KeyValue::new(
            "test-metric-attribute-key",
            "test-metric-attribute-value",
        )]),
    );
```

Reply:
Hmm, I guess that's for a slightly different purpose, I don't know if that's the right tool then

Reply:
Okay @ThetaSinner - I've added to the crate-level docs sections for naming conventions and a metric name registry: https://github.com/holochain/holochain/blob/40c8e550b6e8d3ccdc840d4ac1e444a5c063ada6/crates/holochain_metrics/src/lib.rs#L34-L89

May be easier to view by checking it out and running `cargo doc`. There's a lot of opportunity for benchmarking here, and I don't care about naming things, so if you'd like something to be different, feel free to edit it directly.